src/usr/share/netcheck/netcheck.py

```diff
@@ -776,3 +794,55 @@ def _calculate_next_log_time(self, loop_time):
         Returns the datetime that the logging should occur.
         """
         return loop_time + datetime.timedelta(seconds=self.config['periodic_status_delay'])
+
+    def _check_for_gateway_change(self):
+        """Checks the current state of the default gateway and issues a broadcast if it is
+        different from the prior state.
+        """
+
+        new_gateway_state = self._get_default_gateway_state()
+
+        if new_gateway_state != self.default_gateway_state:
+            self.logger.info(
+                'The default gateway has changed to %s via %s on interface %s.',
+                new_gateway_state['address'],
+                new_gateway_state['connection_id'],
+                new_gateway_state['interface'])
+
+            self.default_gateway_state = new_gateway_state
+            self.broadcaster.issue()
+
+    def _get_default_gateway_state(self):
+        """Retrieves information about the current default gateway.
+
+        Returns a dictionary containing a gateway IP address, interface name, and associated
+        connection id.
+        """
+        gateway_state = None
+
+        with pyroute2.IPRoute() as ip_route:
+            default_routes = ip_route.get_default_routes()
+            if default_routes:
+                gateway_state = {
+                    'address': None,
+                    'interface': None,
+                    'connection_id': None}
+                default_route = default_routes[0]
+                gateway_state['address'] = default_route['attrs'] \
+                    [IPROUTE_ATTR_RTA_GATEWAY][IPROUTE_ATTR_VALUE]
+                # Output interfaces are stored as index values, which are conveniently
+                #   represented in an index value that's off by one.
+                output_interface_index = default_route['attrs'][IPROUTE_ATTR_RTA_OIF] \
+                    [IPROUTE_ATTR_VALUE] - 1
+                output_interface_attrs = ip_route.get_links()[output_interface_index] \
+                    ['attrs']
+                gateway_state['interface'] = output_interface_attrs \
+                    [IPROUTE_ATTR_IFLA_IFNAME][IPROUTE_ATTR_VALUE]
```

Comment:
This is a lot better, but I would rather these constant variable names were easier to read. I'd rather have English readable constants versus a constants that match pyroute2.

Reply:
Specifically, the thing that is unclear to me is what "RTA" stands for.

Reply:
We agreed to use English readable constants with the pyroute2 constant name in a comment above each one.